In [1]:
import pandas as pd
import numpy as np

## Analyzing Income Distribution and Household Demographics:

### Part - B(Practical)

In [4]:
dataset = pd.read_csv("household_dataset_200_rows.csv")
dataset

Household_ID  Age_of_Household_Head  Household_Income Education_Level  \
0        HH_0001                     63             88530        Graduate   
1        HH_0002                     53            108557       Secondary   
2        HH_0003                     39             76087        Graduate   
3        HH_0004                     67             83840       Secondary   
4        HH_0005                     32            133451        Graduate   
..           ...                    ...               ...             ...   
195      HH_0196                     61             31896       Post-Grad   
196      HH_0197                     57            145798       Secondary   
197      HH_0198                     66             20237        Graduate   
198      HH_0199                     68             35056       Secondary   
199      HH_0200                     48            129548        Graduate   

     Family_Size Owns_House Urban_Rural  
0              1        Yes       Urban  
1              5        Yes       Rural  
2              1        Yes       Urban  
3              2        Yes       Urban  
4              1        Yes       Urban  
..           ...        ...         ...  
195            3        Yes       Urban  
196            7        Yes       Urban  
197            4        Yes       Urban  
198            5        Yes       Urban  
199            2        Yes       Urban  

[200 rows x 7 columns]

### Q 1 Types Of Data:
#### Identify which columns are categorical and which are numerical.

In [16]:
print(dataset.dtypes)
print("============================================================================")
categorical = dataset.select_dtypes(include=['object', 'category']).columns.tolist()
print(f"Categorical Data : {categorical}")
print("============================================================================")
numerical = dataset.select_dtypes(include=['int64']).columns.tolist()
print(f"Numerical Data : {numerical}")

Household_ID             object
Age_of_Household_Head     int64
Household_Income          int64
Education_Level          object
Family_Size               int64
Owns_House               object
Urban_Rural              object
dtype: object
Categorical Data : ['Household_ID', 'Education_Level', 'Owns_House', 'Urban_Rural']
Numerical Data : ['Age_of_Household_Head', 'Household_Income', 'Family_Size']


Conclusion : Hence All The Columns Are Classified as which Are Categorical And Which Are Numerical.

### Q 2 Central Tendency:
#### Calculate Mean, Median, and Mode of Household Income and Age.
#### Interpret the central tendency and comment on income distribution.

In [48]:
#Household Income
#Mean
sum_values_of_household_income = np.sum(dataset["Household_Income"])
mean = sum_values_of_household_income/len(dataset)
print(f"Mean : {mean}")

#Median
sorted_dataset = np.sort(dataset["Household_Income"])
first_part = len(sorted_dataset)//2
first_part_value = sorted_dataset[first_part]

second_part = first_part - 1
second_part_value = sorted_dataset[second_part]
median = (first_part_value + second_part_value)/2
print(f"Median : {median}")

#Mode
frequency = {}
for value in dataset["Household_Income"]:
    if value in frequency:
        frequency[value] += 1
    else:
        frequency[value] = 1

max_count = max(frequency.values())
modes = [key for key, count in frequency.items() if count == max_count]

print(f"Value : {modes[0]}")
print(f"Appears: {max_count} times")


Mean : 86959.77
Median : 89293.5
Value : 88530
Appears: 1 times


In [ ]:
#Age
#Mean
sum_values_of_household_income = np.sum(dataset["Age_of_Household_Head"])
mean = sum_values_of_household_income/len(dataset)
print(f"Mean : {mean}")

#Median
sorted_dataset = np.sort(dataset["Age_of_Household_Head"])
first_part = len(sorted_dataset)//2
first_part_value = sorted_dataset[first_part]

second_part = first_part - 1
second_part_value = sorted_dataset[second_part]
median = (first_part_value + second_part_value)/2
print(f"Median : {median}")

#Mode
frequency = {}
for value in dataset["Age_of_Household_Head"]:
    if value in frequency:
        frequency[value] += 1
    else:
        frequency[value] = 1

max_count = max(frequency.values())

modes = [key for key, count in frequency.items() if count == max_count]
print(f"Value : {modes[0]}")
print(f"Appears: {max_count} times")

Mean : 49.26
Median : 50.0
Value : 68
Appears: 10 times


### Central Tendency : HouseHold
Mean : 86959.77

Median : 89293.5

Mode : 88530

### Central Tendency : Age
Mean : 49.26

Median : 50.0

Mode : 68

### Interpretation
1)Since We Can See That In Household Income Mean is Greater Than Median Which Mean The Data Is Positively Skewed (Right Skewed Distribution).
Which means Only Few Household Earns More Income Than All Others. 

2)And We can See that Age Column Is perfectly Symmetric (Normal Distribution).

### Comment on Income Distribution
The Income Distribution Is Not Symmetric. It Is Positively Skewed .

Hence,We Can say That Above Given All Mean,Median,Mode Are Calculated for Both Household Income And Age.

### Q 3 Measures of Dispersion:
#### Calculate Range, Variance, and Standard Deviation of Household Income.
#### Find IQR (Interquartile Range) and comment on income spread.

In [68]:
#Range
range = max(dataset["Household_Income"]) - min(dataset["Household_Income"])
print(f"Range : {range}")

#Variance
deviations = 0
for i in dataset["Household_Income"]:
    deviations += (i - mean)**2

sample_variance = deviations/(len(dataset)-1)
population_variance = deviations/len(dataset)
print(f"Sample Variance : {sample_variance}")
print(f"Population Variance : {population_variance}")

#Standard Deviation
std = np.sqrt(population_variance)
print(f"Standard Deviation : {std}")

Range : 132140
Sample Variance : 1673840946.2885435
Population Variance : 1665471741.5571008
Standard Deviation : 40810.19163832854


In [ ]:
#InterQuarter Range:


# Q 4 